api/v1alpha1/types_etcd.go

```diff
@@ -178,6 +178,13 @@ type BackupSpec struct {
 	// DeltaSnapshotMemoryLimit defines the memory limit after which delta snapshots will be taken
 	// +optional
 	DeltaSnapshotMemoryLimit *resource.Quantity `json:"deltaSnapshotMemoryLimit,omitempty"`
+	// DeltaSnapshotRetentionPeriod defines the duration for which delta snapshots will be retained, excluding the latest snapshot set.
+	// The value should be a string formatted as a duration (e.g., '1s', '2m', '3h', '4d') with a minimum value of 1 second.
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([1-9][0-9]*([.][0-9]+)?(s|m|h|d))+$"
```

Comment:
Thanks for the additional validation for the value.
Let's look at the following cases:
1. `0s`, `0d` (zero-values) - not allowed. Please check previous comment for why these values should be allowed.
2. `5d0h`, `2h0m` (non-zero values with zero-value items) - not allowed. These are perfectly valid durations, according to [metav1.Duration code](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/duration_test.go), and should be allowed.
3. `0.5d`, `0.8h` (fractional values less than 1) - not allowed. These could be valid usecases where user would like to set fractional values.
4. `5s6s7s` - allowed. Does this result in final duration value of `18s`, ie, are the elements summed up as is without checking if a certain time-unit is repeated?

Overall, I would prefer if zero-value elements are allowed in the value, to conform with standard metav1.Duration. And since etcd-backup-restore also allows the value to be zero (by default too), it's ok to allow setting this in the `Etcd` resource as well.

Reply:
Yeah Shreyas, permitting zero values does make sense. Thank you for the suggestion.
 Fixed as part of 4f0adfd3eac2c2011c5fb34ace6c83ffc128adf9